### PR TITLE
Add tools notebook

### PR DESCRIPTION
**Pull request by anakin87**
### Related Issues

- fixes https://github.com/deepset-ai/haystack/issues/8312

### Proposed Changes:

Add a notebook covering the major changes introduced for experimental Tools support

### Notes for the reviewer

Notebook preview: https://github.com/deepset-ai/haystack-experimental/blob/de034e02d7939cb1bb4e434ee8348de4666b20a2/examples/tools_support.ipynb

- The notebook works with `haystack-ai==2.5.0` and `haystack-experimental` main branch (https://github.com/deepset-ai/haystack-experimental/commit/42bfbca02c6652bf71742b169eaa4424e6188804)
- I was unable to use the `ChatMessageRetriever` (I will open an issue with the encountered problems), so I am using a custom component (`MessageCollector`).
- Compared to other ideas recently discussed with Bilge, I decided to keep the `ToolInvoker` very simple after discussing it with Silvano. (This means that a `ConditionalRouter` is often needed in pipelines)

**Since I may not be available or responsive in the next few days, feel free to make changes to this branch.**

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack/blob/main/code_of_conduct.txt)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.
- I documented my code
- I ran [pre-commit hooks](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md#installation) and fixed any issue

### CODE REVIEW

**Comment by TuanaCelik** [comment]
@anakin87 can we make the example a cookbook in the haystack-cookbook repo? You can simply add a copy of it there too.
As for the issues you had and opened here, we now have the discussion board for each experimental feature 🙏

**Review by TuanaCelik — CHANGES_REQUESTED** [review]
@anakin87 can we make the example a cookbook in the haystack-cookbook repo? You can simply add a copy of it there too.
As for the issues you had and opened here, we now have the discussion board for each experimental feature 🙏

**Comment by anakin87** [comment]
> @anakin87 can we make the example a cookbook in the haystack-cookbook repo? You can simply add a copy of it there too. As for the issues you had and opened here, we now have the discussion board for each experimental feature 🙏

I see we have the example notebooks in this repository, too...
Once this notebook is approved and merged, I will copy it into the cookbook. I will also open a related discussion here.
Is this approach ok?

**Comment by TuanaCelik** [comment]
> > @anakin87 can we make the example a cookbook in the haystack-cookbook repo? You can simply add a copy of it there too. As for the issues you had and opened here, we now have the discussion board for each experimental feature 🙏
> 
> 
> 
> I see we have the example notebooks in this repository, too...
> 
> Once the current PR is approved and merged, I will copy the notebook into the cookbook. I will also open a related discussion here.
> 
> Is this approach ok?

Deal,
I'll review asap

File: examples/tools_support.ipynb [hunk 1]
```diff
@@ -0,0 +1,7979 @@
...
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "l68Urk97Wmut"
+      },
+      "source": [
+        "# 🛠️ Tools support\n",
+        "\n",
+        "<a href=\"https://colab.research.google.com/github/deepset-ai/haystack-experimental/blob/main/examples/tools_support.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\" width=\"200\"/></a>\n",
+        "\n",
+        "In this notebook, we introduce the experimental features we've been developing for tool/function calling support in Haystack.\n",
```
Comment (TuanaCelik) [inline examples/tools_support.ipynb:13]: Add something about what they will be building

File: examples/tools_support.ipynb [hunk 2]
```diff
@@ -0,0 +1,7979 @@
...
+        "- We introduced the `ToolInvoker` component, to actually execute tool calls prepared by Language Models."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "toc",
+        "id": "sxtzQK0CWubE"
+      },
+      "source": [
```
Comment (TuanaCelik) [inline examples/tools_support.ipynb:29]: Remove These TOCs that have `>> ` at the beginning because they will look weird in the cookbooks on the website

**Review by TuanaCelik — APPROVED** [review]
Only 2 very minor comments, the cookbook is great, I understood each step, and I like the additions to the chat message dataclass! Thanks @anakin87

File: examples/tools_support.ipynb (hunk 1 above)
Comment (anakin87) [reply examples/tools_support.ipynb:13]: done

File: examples/tools_support.ipynb (hunk 2 above)
Comment (anakin87) [reply examples/tools_support.ipynb]: done

**Pull request merged**